lib/bibliothecary/parsers/maven.rb

```diff
@@ -141,16 +141,30 @@ def self.parse_gradle(manifest)
         end.compact
       end
 
-      def self.extract_pom_dep_info(xml, dependency, name)
+      def self.extract_pom_info(xml, location, parent_properties = {})
+        extract_pom_dep_info(xml, xml, location, parent_properties)
+      end
+
+      def self.extract_pom_dep_info(xml, dependency, name, parent_properties = {})
         field = dependency.locate(name).first
         return nil if field.nil?
         value = field.nodes.first
         match = value.match(/^\$\{(.+)\}/)
         if match
-          return value unless xml.respond_to? 'properties'
+          # the xml root is <project> so lookup the non property name in the xml
+          # this converts ${project/group.id} -> ${group/id}
+          non_prop_name = match[1].gsub('.', '/').gsub('project/', '')
+          return value if !xml.respond_to?('properties') && parent_properties.empty? && !xml.locate(non_prop_name)
           prop_field = xml.properties.locate(match[1]).first
+          parent_prop = parent_properties[match[1]]
           if prop_field
             return prop_field.nodes.first
+          elsif parent_prop
+            return parent_prop
+          elsif xml.locate(non_prop_name).first
```

Comment:
I've added another type of lookup here to expand variables that are not set as properties. An example would be using ${project.groupId}. That won't be defined under <properties> in the pom, but we would still want to look it up. Let me know your thoughts on this. I've added it to the Maven parse spec.

Reply:
I think this makes sense